src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -561,8 +572,20 @@ class ChooseStringGenerator : public Generator<std::string> {
             // If not passed in, give each choice equal weight
             _weights.assign(_choices.size(), 1);
         }
+        if (node["deterministic"]) {
+            auto val = node["deterministic"].maybe<int32_t>().value_or(-1);
```

Comment:
Would it make sense for 'deterministic' to be a boolean value?

Reply:
Done.